src/utils.ts

```diff
@@ -18,17 +28,45 @@ export const getSubtypeAndSridFromModifier = (
   if (!isGeography && srid !== 0) {
     throw new Error("Unexpected SRID with geometry type");
   }
-  const subtype = (modifier & 255) >> 2;
-  if (subtype > 7 || subtype < 0) {
+  if (
+    subtype !== 0 &&
+    subtype !== 1 &&
+    subtype !== 2 &&
+    subtype !== 3 &&
+    subtype !== 4 &&
+    subtype !== 5 &&
+    subtype !== 6 &&
+    subtype !== 7
+  ) {
     throw new Error(
       `Unsupported PostGIS modifier, expected 0-7, received ${subtype} (${modifier})`
     );
   }
-  const isXym = (modifier & 1) === 1;
+
   return {
-    srid,
-    isXym,
     subtype,
     subtypeString: TYPE_LOOKUP[subtype],
+    hasZ,
+    hasM,
+    srid,
   };
 };
+
+// Ref: https://github.com/postgis/postgis/blob/2.5.2/liblwgeom/liblwgeom.h.in#L156-L173
+// #define TYPMOD_SET_SRID(typmod, srid) ((typmod) = (((typmod) & 0xE00000FF) | ((srid & 0x001FFFFF)<<8)))
+// #define TYPMOD_SET_TYPE(typmod, type) ((typmod) = (typmod & 0xFFFFFF03) | ((type & 0x0000003F)<<2))
+// #define TYPMOD_SET_Z(typmod) ((typmod) = typmod | 0x00000002)
+// #define TYPMOD_SET_M(typmod) ((typmod) = typmod | 0x00000001)
+export const getGisTypeModifier = (
+  subtype: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7,
+  hasZ: boolean,
+  hasM: boolean,
+  srid: number // We only support SRID 4326 currently
+): number => {
+  return (
+    ((srid & 0x001fffff) << 8) +
+    ((subtype & 0x0000003f) << 2) +
+    (hasZ ? 0x00000002 : 0) +
+    (hasM ? 0x00000001 : 0)
```

Comment:
I assume this is from the PostGIS source too? If so, feel free to merge (but maybe add a comment like before).

Reply:
It misplaced above the function. Fixed in add22e1